core/src/main/java/org/testcontainers/images/RemoteDockerImage.java

```diff
@@ -95,4 +92,21 @@ protected final String resolve() {
             throw new ContainerFetchException("Failed to get Docker client for " + imageName, e);
         }
     }
+
+    private DockerImageName getImageName() throws InterruptedException, ExecutionException {
+        return imageNameFuture.get();
+    }
+
+    @ToString.Include(name = "imageName", rank = 1)
+    private String imageNameToString() {
+        if (!imageNameFuture.isDone()) {
+            return "<resolving>";
+        }
+
+        try {
+            return getImageName().toString();
+        } catch (InterruptedException | ExecutionException e) {
+            return e.getMessage();
```

Comment:
This caught me by surprise - in case of a failure, why would we not rethrow? At first sight it seems odd that an exception message could become the 'name' of the image...

Reply:
this is for `toString()` - it would be weird to rethrow from `toString` IMO. Instead, we return the message, so that `RemoveDockerImage#toString` shows that there was an exception

Reply:
Fair enough - I guess we can't do too much more than this.

Reply:
Yes, might create some nasty bugs if rethrowing.